src/V3EmitC.cpp

```diff
@@ -4108,11 +4127,15 @@ class EmitCTrace final : EmitCStmts {
 
 static void setParentClassPointers() {
     // Set user4p in all CFunc and Var to point to the containing AstNodeModule
-    for (AstNode* modp = v3Global.rootp()->modulesp(); modp; modp = modp->nextp()) {
+    auto setAll = [](AstNodeModule* modp) -> void {
         for (AstNode* nodep = VN_CAST(modp, NodeModule)->stmtsp(); nodep; nodep = nodep->nextp()) {
             if (VN_IS(nodep, CFunc) || VN_IS(nodep, Var)) nodep->user4p(modp);
         }
+    };
+    for (AstNode* modp = v3Global.rootp()->modulesp(); modp; modp = modp->nextp()) {
+        setAll(VN_CAST(modp, NodeModule));
```

Comment:
It's a separate function because it is called one more time just below the loop on the const pool module. Written as a lambda because C++ does not support basic nested function definitions and I dislike adding more and more functions in file scope that are only called from within one function as a helper. This way you don't wonder where else that trivial utility function is used. Coming from a functional programming language, the helper with restricted scope is easier to read for me than yet another global function with static linkage, but I am happy not to use this style if you have a strong preference.

Reply:
IMO it's the coder's prerogative in this case so it's fine.